subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -558,23 +743,99 @@ uint8_t ull_adv_sync_pdu_set_clear(struct ll_adv_set *adv,
 		}
 	}
 
-	err = adv_sync_hdr_set_clear(lll_sync, pdu_prev, pdu_new,
+#if defined(CONFIG_BT_CTLR_ADV_EXT_PDU_EXTRA_DATA_MEMORY)
+	if (extra_data_prev) {
+		*extra_data_prev = ed_prev;
+	}
+	if (extra_data_new) {
+		*extra_data_new = ed_new;
+	}
+#endif /* CONFIG_BT_CTLR_ADV_EXT_PDU_EXTRA_DATA_MEMORY */
+
+	*ter_pdu_prev = pdu_prev;
+	*ter_pdu_new = pdu_new;
+
+	return 0;
+}
+
+/* @brief Set or clear fields in extended advertising header and store
+ *        extra_data if requested.
+ *
+ * @param[in]  adv              Advertising set.
+ * @param[in]  hdr_add_fields   Flag with information which fields add.
+ * @param[in]  hdr_rem_fields   Flag with information which fields remove.
+ * @param[in]  data             Pointer to data to be added to header and
+ *                              extra_data. Content depends on the value of
+ *                              @p hdr_add_fields.
+ * @param[out] ter_idx          Index of new PDU.
+ *
+ * @Note
+ * @p data content depends on the flag provided by @p hdr_add_fields:
+ * - ULL_ADV_PDU_HDR_FIELD_CTE_INFO:
+ *   # @p data->field_data points to single byte with CTEInfo field
+ *   # @p data->extra_data points to memory where is struct lll_df_adv_cfg
+ *     for LLL.
+ * - ULL_ADV_PDU_HDR_FIELD_AD_DATA:
+ *   # @p data->field_data points to memory where first byte
+ *     is size of advertising data, following byte is a pointer to actual
+ *     advertising data.
+ *   # @p data->extra_data is NULL
+ * - ULL_ADV_PDU_HDR_FIELD_AUX_PTR: # @p data parameter is not used
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+uint8_t ull_adv_sync_pdu_set_clear(struct lll_adv_sync *lll_sync,
+				   struct pdu_adv *ter_pdu_prev,
+				   struct pdu_adv *ter_pdu,
+				   uint16_t hdr_add_fields,
+				   uint16_t hdr_rem_fields,
+				   struct adv_pdu_field_data *data)
+{
+	int err;
+
+	err = adv_sync_hdr_set_clear(lll_sync, ter_pdu_prev, ter_pdu,
```

Comment:
It looks that `adv_sync_hdr_set_clear` is not needed anymore. Signature of the function `ull_adv_sync_pdu_set_clear` may be changed to use `void* *data` instead of `struct adv_pdu_field_data *data`.